jobs: Add created by fields to job record.

miretskiy opened the pull request:
Make it possible for the job creator to specify `created_by_type`
and `created_by_id` fields, which are now stored in the `system.jobs` table.

Release Notes: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/51554)
<!-- Reviewable:end -->

spaskob reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/51554#-:-MCXAs_W9m6E37na-TkG:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dt and @spaskob)



<!-- Sent from Reviewable.io -->

Commit:
jobs: Add created by fields to job record.

Make it possible for the job creator to specify
`created_by_type` and `created_by_id` fields, which are now
stored in the `system.jobs` table.

Release Notes: None

miretskiy commented:
bors r+
